src/code_8069E0C.c

```diff
@@ -295,39 +337,1982 @@ void nullsub_95(void)
+            }
+
+            if (j == count) {
+                SetSpeciesLevelToExtract(loopPtr, 0, monStructPtr->speciesNum);
+                unk2Field[0] = 0;
+                unk2Field[1] = 0;
+                unk2Field += sizeof(*unkPtr) / sizeof(unkPtr->unk2[0]); // Hacky solution, because loopPtr->unk2[0/1] = 0; doesn't match
+                loopPtr++;
+                count++;
+            }
+        }
+    }
+
+    count = sub_803D808(unkPtr, count);
+    if (a1) {
+        SetSpeciesLevelToExtract(&unkPtr[count], 1, MONSTER_DECOY);
+        unkPtr[count].unk2[0] = 0;
+        unkPtr[count].unk2[1] = 0;
+        count++;
+    }
+
+    return count;
+}
+
+extern const DungeonPos gUnknown_80F4598[];
+
+void sub_806B168(void)
+{
+    s32 count;
+    PokemonStruct2 *monPtrs[MAX_TEAM_MEMBERS];
+    PokemonStruct2 *monPtrs2[MAX_TEAM_MEMBERS]; // Leader and partner?
+    u8 playerSpawnRoomId = GetTile(gDungeon->playerSpawn.x, gDungeon->playerSpawn.y)->room;
+    s32 i;
+
+    for (i = 0; i < MAX_TEAM_MEMBERS; i++) {
+        PokemonStruct2 *currMonPtr = &gRecruitedPokemonRef->pokemon2[i];
+        currMonPtr->unkC = i;
+    }
+
+    // Populate monPtrs
+    count = 0;
+    for (i = 0; i < MAX_TEAM_MEMBERS; i++) {
+        PokemonStruct2 *currMonPtr = &gRecruitedPokemonRef->pokemon2[i];
+        if (PokemonFlag1Struct2(currMonPtr) && PokemonFlag2Struct2(currMonPtr)) {
+            monPtrs[count++] = currMonPtr;
+        }
+    }
+    for (; count < MAX_TEAM_MEMBERS; count++) {
+        monPtrs[count] = NULL;
+    }
+
+    // Populate monPtrs2
+    count = 0;
+    for (i = 0; i < MAX_TEAM_MEMBERS; i++) {
+        PokemonStruct2 *currMonPtr = monPtrs[i];
+        if (currMonPtr != NULL && currMonPtr->isTeamLeader) {
+            monPtrs[i] = NULL;
+            monPtrs2[count++] = currMonPtr;
+        }
+    }
+
+    if (gDungeon->unk644.unk18 == 0) {
+        for (i = 0; i < MAX_TEAM_MEMBERS; i++) {
+            PokemonStruct2 *currMonPtr = monPtrs[i];
+            if (currMonPtr != NULL && currMonPtr->dungeonLocation.id == DUNGEON_JOIN_LOCATION_PARTNER) {
+                monPtrs[i] = NULL;
+                monPtrs2[count++] = currMonPtr;
+            }
+        }
+    }
+
+    for (i = 0; i < MAX_TEAM_MEMBERS; i++) {
+        PokemonStruct2 *currMonPtr = monPtrs[i];
+        if (monPtrs[i] != NULL) {
+            monPtrs[i] = NULL;
+            monPtrs2[count++] = currMonPtr;
+        }
+    }
+    for (; count < MAX_TEAM_MEMBERS; count++) {
+        monPtrs2[count] = NULL;
+    }
+
+    // Do something else
+    for (i = 0; i < MAX_TEAM_MEMBERS; i++) {
+        PokemonStruct2 *currMonPtr = monPtrs2[i];
+
+        if (currMonPtr != NULL) {
+            DungeonPos unkPosition;
+            s32 j;
+            const Tile *tile;
+            bool8 skipSecondLoop;
+
+            if (currMonPtr->speciesNum == MONSTER_CASTFORM_SNOWY || currMonPtr->speciesNum == MONSTER_CASTFORM_RAINY || currMonPtr->speciesNum == MONSTER_CASTFORM_SUNNY) {
```

Comment:
Can this use the new `IS_CASTFORM_FORM_MONSTER` macro?

Reply:
It sadly can't be used here, as the species are in different order which changes asm generation.